### PR TITLE
Fix comparison using nullable objects

### PR DESCRIPTION
**Pull request by RodrigoDLopez**
## Description

This PR fix some wrongs comparison using nullable objects. Preventing null pointer exception

## Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

### CODE REVIEW

File: utils/src/test/java/com/cloud/utils/TernaryTest.java [hunk 1]
```diff
@@ -28,7 +28,7 @@ public void testEquals() {
         Assert.assertEquals(new Ternary<String, String, String>("a", "b", "c"), new Ternary<String, String, String>("a", "b", "c"));
         Assert.assertFalse(new Ternary<String, String, String>("a", "b", "c").equals(new Ternary<String, String, String>("a", "b", "d")));
         Assert.assertFalse(new Ternary<String, String, String>("a", "b", "c").equals(""));
-        Assert.assertFalse(new Ternary<String, String, String>("a", "b", "c").equals(null));
```
Comment (rafaelweingartner) [inline utils/src/test/java/com/cloud/utils/TernaryTest.java:31]: You do not need to remove the `equals` here. In this very specific use case, there is no null pointer that can happen here.

File: engine/orchestration/src/test/java/com/cloud/agent/manager/ConnectedAgentAttacheTest.java [hunk 2]
```diff
@@ -44,7 +44,7 @@ public void testEqualsFalseNull() throws Exception {
 
         ConnectedAgentAttache agentAttache1 = new ConnectedAgentAttache(null, 0, null, link, false);
 
-        assertFalse(agentAttache1.equals(null));
```
Comment (rafaelweingartner) [inline engine/orchestration/src/test/java/com/cloud/agent/manager/ConnectedAgentAttacheTest.java:47]: The same here. I do not understand why people coded this check, but it does not cause a null pointer. Moreover, if you change the equals to `==` here, you are changing the test case.

File: server/src/main/java/com/cloud/storage/VolumeApiServiceImpl.java [hunk 3]
```diff
@@ -2199,7 +2199,7 @@ public Volume migrateVolume(MigrateVolumeCmd cmd) {
         // OfflineVmwareMigration: check storage tags on disk(offering)s in comparison to destination storage pool
         // OfflineVmwareMigration: if no match return a proper error now
         DiskOfferingVO diskOffering = _diskOfferingDao.findById(vol.getDiskOfferingId());
-        if (diskOffering.equals(null)) {
+        if (diskOffering == null) {
```
Comment (rafaelweingartner) [inline server/src/main/java/com/cloud/storage/VolumeApiServiceImpl.java:2202]: This is the only one that makes sense to change.

Comment (RodrigoDLopez) [reply server/src/main/java/com/cloud/storage/VolumeApiServiceImpl.java:2202]: thanks for the hint, I did not know about this other specific use case.

File: utils/src/test/java/com/cloud/utils/TernaryTest.java (hunk 1 above)
Comment (RodrigoDLopez) [reply utils/src/test/java/com/cloud/utils/TernaryTest.java]: thanks for the hint, I did not know about this other specific use case.

**Comment by RodrigoDLopez** [comment]
@blueorangutan package

**Comment by rhtyd** [comment]
@blueorangutan test

**Comment by rhtyd** [comment]
@blueorangutan test

**Pull request merged**